src/Game.java

```diff
@@ -3,6 +3,7 @@
 public final class Game {
     private Board board;
     private ArrayList<Player> players;
+    private Player currentPlayer;
```

Comment:
What if we represent the current player here by its index? We can think of it like an "id" and could reasonbly call it `currentPlayerId`.

#### Why?
It also allows the `Player` objects to be stored in a **single place**, the `players` list. It's typically a good idea to minimize the number of persistent references to objects. What would happen if we swap `currentPlayer` out of the list without updating the value of currentPlayer? Our game would be in an inconsistent state. Using an id here provides better consistency.